sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/EquivalentExpressions.scala

```diff
@@ -229,3 +208,27 @@ class EquivalentExpressions {
     sb.toString()
   }
 }
+
+/**
+ * Orders `Expression` by parent/child relations. The child expression is smaller
+ * than parent expression. If there is child-parent relationships among the subexpressions,
+ * we want the child expressions come first than parent expressions, so we can replace
+ * child expressions in parent expressions with subexpression evaluation. Note that
+ * this is not for general expression ordering. For example, two irrelevant expressions
+ * will be considered as e1 < e2 and e2 < e1 by this ordering. But for the usage here,
```

Comment:
Shall we change this to `0` according to the new logic?

Reply:
Right, missing the doc. Fixed.